src/driver/postgres/PostgresDriver.ts

```diff
@@ -1051,7 +1051,10 @@ export class PostgresDriver implements Driver {
      * Closes connection pool.
      */
     protected async closePool(pool: any): Promise<void> {
-        await Promise.all(this.connectedQueryRunners.map(queryRunner => queryRunner.release()));
+        while (this.connectedQueryRunners.length) {
```

Comment:
can you please explain why this change to `while` is needed? As I understand main motivation is to release them in sequence. If so, why its needed? Thanks!

Reply:
Because the `.release()` method of the query runners mutates the driver's `connectedQueryRunners` list, causing `.map()` to not iterate through all of them. I changed it to a while loop because it was the most obvious way i could think to ensure that all query runners got released and `connectedQueryRunners` was really empty before closing the pool 😅. If you have a better way to do it, let me know!